XWSBack/BaseApi.Services/UserServices/EditUserService.cs

```diff
@@ -18,11 +20,41 @@ public EditUserService(IMongoClient client)
         public async Task EditBasicInformations(Guid userId, User newUser)
         {
             var currentUser = await _usersCollection.Find(x => x.Id == userId).FirstOrDefaultAsync();
+            if (currentUser == null)
+                throw new BadRequestException("User doesn't exist!");
+
             var editedUser = EditUser(currentUser, newUser);
 
             await _usersCollection.ReplaceOneAsync(x => x.Id == editedUser.Id, editedUser);
         }
 
+        public async Task AddMilestone(Guid userId, Milestone milestone)
+        {
+            var user = await _usersCollection.Find(x => x.Id == userId).FirstOrDefaultAsync();
+            if (user == null)
+                throw new BadRequestException("User doesn't exist!");
+
+
+            if (user.Experiences == null)
+                user.Experiences = new List<Milestone>();
+
+            user.Experiences.Add(milestone);
+            await _usersCollection.ReplaceOneAsync(x => x.Id == userId, user);
+        }
+
+        public async Task RemoveMilestone(Guid userId, string title, DateTime startTime)
```

Comment:
Da li je dobar nacin da se gleda startTime kao kljuc? Zasto ne bi korisnik mogao da ima dva milestone-a koji su zapoceli u isto vreme? 
Iako nema ključ u sebi ja bih koristio redni broj. Ako neko ukljanja 1, 2, 3 milestone to se moze lakse hendlati.